src/main/scala/com/danielasfregola/twitter4s/http/serializers/JsonSupport.scala

```diff
@@ -30,21 +30,25 @@ private[twitter4s] trait JsonSupport extends Json4sSupport {
 
 private[twitter4s] object DateTimeFormatter {
 
-  val formatterPattern = "EEE MMM dd HH:mm:ss ZZZZ yyyy"
+  val apiV1InstantFormatterPattern = "EEE MMM dd HH:mm:ss ZZZZ yyyy"
   val locale = Locale.ENGLISH
 
   // SimpleDateFormat not thread safe! Damn you java!!!
-  def formatter = new SimpleDateFormat(formatterPattern, locale)
+  def formatter = new SimpleDateFormat(apiV1InstantFormatterPattern, locale)
+
+  val v1InstantFormatter = (s: String) => formatter.parse(s).toInstant
+  val v2InstantFormatter = (s: String) => Instant.parse(s)
 
   val builder: DateTimeFormatterBuilder = new DateTimeFormatterBuilder()
     .appendPattern("[EEE MMM dd HH:mm:ss Z yyyy]")
     .appendPattern("[yyyy-MM-dd HH:mm:ss Z]")
 
   val dateTimeFormatter = builder.toFormatter(locale)
 
-  def canParseInstant(s: String): Boolean = Try(parseInstant(s)).isSuccess
-
-  def parseInstant(s: String): Instant = formatter.parse(s).toInstant
+  def parseInstant(s: String): Option[Instant] = Seq(
```

Comment:
What about keeping the same signature def `parseInstant(s: String): Instant` and just do:

```
Try(v1InstanceFormatter(s)).getOrElse(v2InstantFormatter(s))
```

Reply:
Oh neat, I didn't realize you could `.getOrElse` on Try.